crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -1403,25 +1653,40 @@ where
             (peer_count, Instant::now()),
         );
 
-        self.peers.insert(
-            peer_count,
-            LocalPeerView {
-                address: address.get_net_address(),
-                port: address.get_port(),
-                user_agent: "".to_string(),
-                state: PeerStatus::Awaiting,
-                channel: requests_tx,
-                services: ServiceFlags::NONE,
-                _last_message: Instant::now(),
-                kind,
-                address_id: peer_id as u32,
-                height: 0,
-                banscore: 0,
-                transport_protocol: TransportProtocol::V1, // Default to V1, will be updated when peer is ready.
-            },
-        );
+        let peer_view = LocalPeerView {
+            address: address.get_net_address(),
+            port: address.get_port(),
+            user_agent: "".to_string(),
+            state: PeerStatus::Awaiting,
+            channel: requests_tx,
+            services: ServiceFlags::NONE,
+            _last_message: Instant::now(),
+            kind,
+            address_id: peer_id as u32,
+            height: 0,
+            banscore: 0,
+            transport_protocol,
+        };
 
-        self.peer_id_count += 1;
+        self.peers.insert(peer_count, peer_view);
+
+        if insert_peer {
+            // Check if the peer already exists
+            if self.added_peers.values().any(|(_, peer_addr, _)| {
```

Comment:
I would avoid long statements like this, they are usually harder to read. Something like this is usually better (although we do have cases like this in the code :/ )

```rust
let alread_exists = self.added_peers.values()....
if aready_exists { ... }